### PR TITLE
ER-866_whitelisting_providers

### PR DESCRIPTION
**Pull request by shomavg**
Whitelisting providers.

### CODE REVIEW

File: src/Provider/Provider.Web/Configuration/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -71,10 +72,14 @@ public static void AddMvcService(this IServiceCollection services, IHostingEnvir
                 opts.Filters.AddService<PlannedOutageResultFilter>();
 
                 opts.AddTrimModelBinderProvider(loggerFactory);
+                if (EnvironmentNames.GetTestEnvironmentNames().Contains(hostingEnvironment.EnvironmentName.ToUpper()) == false)
```
Comment (dev4valtech) [inline src/Provider/Provider.Web/Configuration/ConfigurationExtensions.cs]: The if condition can be converted into meaningful function with name like "IsProductionEquivalentEnvironment"

File: src/Employer/Esfa.Recruit.Azure.Resources/azuredeploy.json [hunk 2]
```diff
@@ -110,6 +110,10 @@
       "type": "bool",
       "defaultValue": false
     },
+	"WhitelistedProvidersList":{
```
Comment (dev4valtech) [inline src/Employer/Esfa.Recruit.Azure.Resources/azuredeploy.json]: This needs to be added in the parameters file as well.

Comment (shomavg) [reply src/Employer/Esfa.Recruit.Azure.Resources/azuredeploy.json]: ok..will do!

File: src/Provider/Provider.Web/Configuration/ConfigurationExtensions.cs (hunk 1 above)
Comment (shomavg) [reply src/Provider/Provider.Web/Configuration/ConfigurationExtensions.cs]: It's the same functionality used for Employer... ! i will give it a meaningful name.

File: src/Provider/Provider.Web/Controllers/ErrorController.cs [hunk 3]
```diff
@@ -111,6 +116,15 @@ private IActionResult PageNotFound()
         {
             Response.StatusCode = (int)HttpStatusCode.NotFound;
             return View(ViewNames.PageNotFound);
+        }       
+
+        // Blocked provider url for private beta
```
Comment (chrisjensenuk) [inline src/Provider/Provider.Web/Controllers/ErrorController.cs]: We don't need this comment

File: src/Employer/Esfa.Recruit.Azure.Resources/azuredeploy.json (hunk 2 above)
Comment (sachpatel) [reply src/Employer/Esfa.Recruit.Azure.Resources/azuredeploy.json]: Where is this even used? (rhetorical question)

Comment (shomavg) [reply src/Employer/Esfa.Recruit.Azure.Resources/azuredeploy.json]: We provide a list of UKPRN's in the json file.

**Pull request merged**